src/git/store.ml

```diff
@@ -194,17 +194,32 @@ sig
 
   module Ref :
   sig
+    module Packed_refs : Packed_refs.S with module Hash = Hash
+                                        and module Path = Path
+                                        and module FileSystem = FileSystem
+
     type nonrec error =
-      [ error
+      [ Packed_refs.error
+      | error
       | `Invalid_reference of Reference.t ]
 
     val pp_error : error Fmt.t
     val graph_p : t -> dtmp:Cstruct.t -> raw:Cstruct.t -> (Hash.t Reference.Map.t, error) result Lwt.t
     val graph : t -> (Hash.t Reference.Map.t, error) result Lwt.t
     val normalize : Hash.t Reference.Map.t -> Reference.head_contents -> (Hash.t, error) result Lwt.t
-    val list_p : t -> dtmp:Cstruct.t -> raw:Cstruct.t -> ((Reference.t * Hash.t) list, error) result Lwt.t
-    val list_s : t -> ((Reference.t * Hash.t) list, error) result Lwt.t
-    val list : t -> ((Reference.t * Hash.t) list, error) result Lwt.t
+    val list_p : t -> dtmp:Cstruct.t -> raw:Cstruct.t -> (Reference.t * Hash.t) list Lwt.t
+    val list_s : t -> (Reference.t * Hash.t) list Lwt.t
+    val list : t -> (Reference.t * Hash.t) list Lwt.t
+    val remove_p : t -> dtmp:Cstruct.t -> raw:Cstruct.t -> ?locks:Lock.t -> Reference.t -> (unit, error) result Lwt.t
```

Comment:
why do you need intermediate buffers to remove a reference?

Reply:
In this case, I check if the reference is `packed` or not. So, I need to decode the `packed-refs` file. One other solution is to _systematically_ `unpack` references when we want to create a Git state - and I think, it could be better about the performance because, at this stage, I need to check all the time where is located the reference (and decode for each operation the `packed-refs` file).